src/avatar_action.cpp

```diff
@@ -797,13 +797,31 @@ bool avatar_action::fire( avatar &you, map &m )
     int shots = 0;
 
     you.moves -= reload_time;
+
+    std::function<void()> pre_fire;
+    std::function<void( int )> post_fire;
+    if( args.mode == TARGET_MODE_TURRET_MANUAL ) {
+        const optional_vpart_position vp = g->m.veh_at( you.pos() );
+        turret_data turret;
+        if( vp && ( turret = vp->vehicle().turret_query( you.pos() ) ) ) {
```

Comment:
Just to be sure: `turret = vp->vehicle().turret_query( you.pos() )` ain't a check, it's an assignment, and yet it's located inside a check. Is it ok?

Reply:
`turret` gets assigned, then checked. I left this shortcut from the original code, it seemed harmless enough